insertSizeMetrics.wdl

```diff
@@ -0,0 +1,84 @@
+version 1.0
+
+workflow insertSizeMetrics {
+  input {
+    File inputBam
+    String outputFileNamePrefix = basename(inputBam, '.bam')
+  }
+
+  call collectInsertSizeMetrics {
+    input:
+      inputBam = inputBam,
+      outputPrefix = outputFileNamePrefix
+  }
+
+  output {
+    File insertSizeMetrics = collectInsertSizeMetrics.insertSizeMetrics
+    File histogramReport = collectInsertSizeMetrics.histogramReport
+  }
+
+  parameter_meta {
+    inputBam: "Input file (bam or sam)."
+    outputFileNamePrefix: "Output prefix to prefix output file names with."
+  }
+
+  meta {
+    author: "Michael Laszloffy"
+    email: "michael.laszloffy@oicr.on.ca"
+    description: "Workflow to run picard InsertSizeMetrics"
+    dependencies: [{
+      name: "picard/2.21.2",
+      url: "https://broadinstitute.github.io/picard/"
+    },{
+      name: "rstats/3.6",
+      url: "https://www.r-project.org/"
+    }]
+  }
+}
+
+task collectInsertSizeMetrics {
+  input {
+    File inputBam
+    String picardJar = "$PICARD_ROOT/picard.jar"
+    Float minimumPercent = 0.5
+    String outputPrefix = "OUTPUT"
+    Int jobMemory = 18
```

Comment:
Unless it is a new way of doing things, I would recommend having all these variables with preset values as optional, as in https://wiki.oicr.on.ca/display/GSI/WDL+Best+Practices#WDLBestPractices-Taskblock

Reply:
These are all optional as they have a default.